docs/proposals/removal-tekton-resources.md

```diff
@@ -0,0 +1,269 @@
+
+For a Git resource, the step that we create in the TaskSpec then looks like the following. Every occurrence of `default` comes from the name of the item in `spec.sources`, or `default` for `spec.source`:
+
+```yaml
+[...]
+taskSpec:
+  steps:
+    - name: source-default
+      image: some-registry/shipwright-io/git-source@sha:08154711
+      args:
+        - --url=https://github.com/shipwright-io/sample-java                 # from the user's Build
+        - --revision=develop                                                 # from the user's Build, omitted if not specified which will load the default branch
+        - --target=$(params.source-root)                                     # resolves to /workspace/source
+        - --result-file-commit-sha=$(results.source-default-commit-sha.path)
+      resources: [...]
+    [...]
+```
+
+For a private repository, we will explicitly mount the secret and not rely on Tekton's credentials initialization (= we will also stop adding this secret to the service account; and will not anymore require a tekton annotation in this secret):
+
+```yaml
+[...]
+taskSpec:
+  steps:
+    - name: source-default
+      image: some-registry/shipwright-io/git-source@sha:19237192
+      args:
+        - --url=git@github.com:shipwright-io/sample-nodejs-private.git       # from the user's Build
+        - --revision=develop                                                 # from the user's Build, omitted if not specified which will load the default branch
+        - --target=$(params.source-root)                                     # resolves to /workspace/source
+        - --result-file-commit-sha=$(results.source-default-commit-sha.path)
+        - --secret-path=/workspace/source-secrets/default
+      resources: [...]
+      volumeMounts:
+        - mountPath: /workspace/source-secrets/default
+          name: ssh-secret                                                   # from the user's build
+          readOnly: true
+  volumes:
+    - name: ssh-secret                                                       # name of the secret
+      secret:
+        secretName: ssh-secret                                               # from the user's build
+        defaultMode: 256                                                     # this is 0400 in octal
+    [...]
+```
+
+The image URL is a new configuration option of the build controller deployment (similar to how Tekton allows to specify its supporting images). In our release process, we need to make sure we point to the right image in our yaml.
+
+The resources of the step are also configurable at the build controller deployment level with reasonable defaults.
+
+The above example assumes that we have one step per source. It is also possible to have one step that handles all sources with advantages for both options:
+
+| Single container | Multiple containers |
```

Comment:
Just as a comment, I _think_ I would prefer container per source.

Reply:
Yep, me too.

Reply:
+1, do we think that's enough agreement to remove "single container" as an option, and move it to some "alternatives considered" section below?

Reply:
Yep, three opinions is good. I will move it to alternatives.